bundle/bundle.go

```diff
@@ -121,10 +121,18 @@ func TryLoad(ctx context.Context) (*Bundle, error) {
 	return Load(ctx, root)
 }
 
+func (b *Bundle) InitializeWorkspaceClient() (*databricks.WorkspaceClient, error) {
+	client, err := b.Config.Workspace.Client()
+	if err != nil {
+		return nil, fmt.Errorf("cannot resolve bundle auth configuration: %w", err)
+	}
+	return client, nil
+}
+
 func (b *Bundle) WorkspaceClient() *databricks.WorkspaceClient {
 	b.clientOnce.Do(func() {
```

Comment:
do we really need `b.clientOnce.Do` still? Now b.client initialised either in bundle mutator or in `MustWorkspaceClient`.
I'd expect we don't use `WorkspaceClient()` outside of these 2 context, so we can simplify it to

```
func (b *Bundle) WorkspaceClient() *databricks.WorkspaceClient {
   if b.client == nil {
     panic("Oh no, not expected")
   }
   return b.client
}

```

Reply:
Done. There was also the databricks labs command group which uses a client without triggering the initialize mutator. Now also calling the initialize function from the databricks labs command.

Reply:
I reverted this. Let's keep `clientOnce`. Removing it is not trivial since a bunch of tests depend upon it. One option would have been to make the `InitialiseWorkspaceClient` a `DefaultMutator()`, but unfortunately target resolution happens after default mutators are applied, and we need to initialize the workspace client after target resolution (because target could contain different auth credentials).

Alternately we could call `InitializeWorkspaceClient` from all tests that call `b.WorkspaceClient()` but that's a bigger lift and arguablely not worth it.

WDYT?